From b9bef8e9555645fc91fab705bec697214a39dbc1 Mon Sep 17 00:00:00 2001
From: Elly <ellyjones@chromium.org>
Date: Thu, 2 Nov 2023 22:46:35 +0000
Subject: [PATCH] atspi: mark possibly unused gn variables

If the atspi version is >= 2.49.90, then gn complains that
major and micro are unused. Removing those variables makes the code less
readable, so instead, mark all three of them as possibly unused.

Bug: None
Change-Id: Ic99c129af9673def54b492b9694063f4b8b6df96
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5001687
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Auto-Submit: Elly FJ <ellyjones@chromium.org>
Commit-Queue: Elly FJ <ellyjones@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1219173}
---
 build/config/linux/atspi2/BUILD.gn | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/build/config/linux/atspi2/BUILD.gn b/build/config/linux/atspi2/BUILD.gn
index d1629205c82a30e..30bc77f9c23fbdd 100644
--- a/build/config/linux/atspi2/BUILD.gn
+++ b/build/config/linux/atspi2/BUILD.gn
@@ -21,10 +21,21 @@ if (use_atk) {
     minor = atspi_version[1]
     micro = atspi_version[2]
 
+    # These aren't necessarily used if atspi is not old enough to require them.
+    # Also, gn considers variables unused if the only use of them is
+    # short-circuited away, so for example if major == 2 and minor == 48, micro
+    # would be unused.
+    not_needed([
+                 "major",
+                 "minor",
+                 "micro",
+               ])
+
     # ATSPI 2.49.90 now defines these for us and it's an error for us to
     # redefine them on the compiler command line.
     # See ATSPI 927344a34cd5bf81fc64da4968241735ecb4f03b
-    if (minor < 49 || (minor == 49 && micro < 90)) {
+    if (major < 2 || (major == 2 && minor < 49) ||
+        (major == 2 && minor == 49 && micro < 90)) {
       defines = [
         "ATSPI_MAJOR_VERSION=$major",
         "ATSPI_MINOR_VERSION=$minor",
